Automattic/portkey-android
# Hide control buttons when editing text

Fix #167 

The problem here was that when editing an existing `ViewType.TEXT` view, two listeners would get triggered: one for the movement which calls `OnPhotoEditorListener.onStartViewChangeListener` when pressing down and `OnPhotoEditorListener.onStopViewChangeListener` when movement stops (finger is lifted), and the effective click listener for TextView so this triggers the TextDialog to be presented.

This is now saved by having a flag so we don't run the call for `OnPhotoEditorListener.onStopViewChangeListener` in the case of a Text type having been tapped on (cc8962c), and also while playing with multitouch I saw we could be triggering multiple TextEditDialogs as tapping on each text view would launch several of them (one per each TextView). Taking advantage of the flag introduced in cc8962c, we now also make a check first so we don't allow editing more than one text at a time (b8ed5c0).

![hidecontrolsonedit](https://user-images.githubusercontent.com/6597771/70722113-78e35e80-1cd5-11ea-9b14-0fe9d0812af8.gif)


To test:
1. open portkey and take a picture
2. tap on the `Tt` icon
3. observe the dialog is presented, and the edit controls are hidden (no cross, no Tt, no Stickers button are shown)
4. enter some text and tap `Done`
5. observe the text is added as a view
6. now tap on the text again, to trigger the editor dialog again
7. observe the edit controls are hidden now (wasn't the case before this PR).
